File: typecast-java/src/main/java/com/neosapience/exceptions/UnprocessableEntityException.java
package com.neosapience.exceptions;

/**
 * Exception thrown when the API returns a 422 Unprocessable Entity error.
 * 
 * <p>This typically indicates a validation error in the request.</p>
 */
public class UnprocessableEntityException extends TypecastException {
    
    /**
     * Creates a new UnprocessableEntityException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public UnprocessableEntityException(String message, String responseBody) {
        super(message, 422, responseBody);
    }

    /**
     * Creates a new UnprocessableEntityException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public UnprocessableEntityException(String message, String responseBody, Throwable cause) {
        super(message, 422, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/UnprocessableEntityException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class UnprocessableEntityException extends TypecastException
{
    public function __construct(string $message = 'Unprocessable entity', ?\Throwable $previous = null)
    {
        parent::__construct($message, 422, $previous);
    }
}
